### PR TITLE
Do not multiply `AggregateRandom` in `JOIN`s

### LINKED ISSUE #11625: Bug Report: vitess generates corrupted results on cross keyspace non full group by queries
### Overview of the Issue

While doing cross keyspace query, vitess generates corrupted results if SQL statement not fulfill full group by constraint. 

### Reproduction Steps

Assume we have 2 keyspaces called `commerce` and `meta`
Running following sql will reproduce this issue.

I think vitess should either raise a error for non-full group by queries if the ONLY_FULL_GROUP_BY sql mode option is on.
Or just return the results as same as mysql default behavior do.

```sql
-- create table for commerce.orders
DROP TABLE IF EXISTS commerce.orders;
CREATE TABLE commerce.orders (
	id INTEGER(10) NOT NULL,
	buyer INTEGER(10) NOT NULL,
	total DECIMAL(10, 2) NOT NULL,
	created DATETIME DEFAULT CURRENT_TIMESTAMP,
	PRIMARY KEY(id)
);


-- create table for meta.users
DROP TABLE IF EXISTS meta.users;
CREATE TABLE meta.users (
	id INTEGER(10) NOT NULL,
	name VARCHAR(64) NOT NULL,
	created DATETIME DEFAULT CURRENT_TIMESTAMP,
	PRIMARY KEY(id)
);

-- create table for commerce.users, this table has the same schema with meta.users table
DROP TABLE IF EXISTS commerce.users;
CREATE TABLE commerce.users (
	id INTEGER(10) NOT NULL,
	name VARCHAR(64) NOT NULL,
	created DATETIME DEFAULT CURRENT_TIMESTAMP,
	PRIMARY KEY(id)
);


-- insert some order into commerce.orders
insert into commerce.orders (id, buyer, total) VALUES(1, 1, 100);
insert into commerce.orders (id, buyer, total) VALUES(2, 1, 200);
insert into commerce.orders (id, buyer, total) VALUES(3, 1, 300);
insert into commerce.orders (id, buyer, total) VALUES(4, 2, 111);
insert into commerce.orders (id, buyer, total) VALUES(5, 2, 222);
insert into commerce.orders (id, buyer, total) VALUES(6, 2, 333);

-- insert some user into meta.users
insert into meta.users (id, name) VALUES(1, 'user 1');
insert into meta.users (id, name) VALUES(2, 'user 2');

-- insert some user into commerce.users
-- now you have 2 users table with same data in both commerce and meta keyspace
insert into commerce.users (id, name) VALUES(1, 'user 1');
insert into commerce.users (id, name) VALUES(2, 'user 2');


-- cross keyspace join with non full group by
-- this query will generated corrupted results
--
-- | id | name | sum(o.total) | created |
-- |----|------|--------------|---------|
-- |  1 |    0 |       600.00 |    2022 |
-- |  2 |    0 |       666.00 |    2022 |
SELECT
	u.id,
	u.name,
	sum(o.total),
	o.created
FROM
	commerce.orders AS o
	JOIN meta.users AS u ON o.buyer = u.id
GROUP BY
	u.id;

-- the same non full group by query, but this time we join tables from the same keyspce
-- this is mysql default behavior
--
-- | id | name   | sum(o.total) | created             |
-- |----|--------|--------------|---------------------|
-- |  1 | user 1 |       600.00 | 2022-10-31 16:39:12 |
-- |  2 | user 2 |       666.00 | 2022-10-31 16:39:51 |
SELECT
	u.id,
	u.name,
	sum(o.total),
	o.created
FROM
	commerce.orders AS o
	JOIN commerce.users AS u ON o.buyer = u.id
GROUP BY
	u.id;

-- fixed query with full group by + proper aggregation statement (min)
--
-- | id | name   | sum(o.total) | created             |
-- |----|--------|--------------|---------------------|
-- |  1 | user 1 |       600.00 | 2022-10-31 16:39:12 |
-- |  2 | user 2 |       666.00 | 2022-10-31 16:39:51 |
SELECT
	u.id,
	u.name,
	sum(o.total),
	min(o.created)
FROM
	commerce.orders AS o
	JOIN meta.users AS u ON o.buyer = u.id
GROUP BY
	u.id, u.name;
```

### Binary Version

```sh
This issue has been tested with

v14.0.1
v14.0.3
v15.0.0

All above versions have the same behavior
```

### Operating System and Environment details

```sh
vitess was setup by vitess k8s operator.
We're using AWS x86_64 ec2 instance with ubuntu 20.20 as k8s worker nodes.
```

### Log Fragments

_No response_

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This Pull Request fixes the issue described in #11625. When pushing aggregation on a join, we were skipping the multiplication of results for `min` and `max` aggregations, however, we also need to skip the multiplication for `random` aggregations. Random aggregations simply need to get the column form one of the join's sides without multiplying it to the other side of the join.

## Related Issue(s)

- Fixes #11625.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by arthurschreiber — APPROVED** [review]
❤️

**Comment by tokikanno** [comment]
@frouioui : A quick question about `AggregateRandom`, does that mean we'll get random rows from non full group by queries?

BTW: 你會說中文? :-P

**Comment by frouioui** [comment]
> A quick question about AggregateRandom, does that mean we'll get random rows from non full group by queries?

@tokikanno, the `AggregateRandom` does not apply a random function to the result, it simply takes the first row that MySQL returns. The results of the `AggregateRandom` will not be "flaky". You can see in the following code snippet how the engine primitive `AggregateRandom` is used:

https://github.com/planetscale/vitess/blob/adeccd2ae1e5b34da0da2fbf89debdc5ccefde2c/go/vt/vtgate/engine/ordered_aggregate.go#L531-L532

> BTW: 你會說中文? :-P

我会一点点 不过我的中文不好 我的大学在北京

**Comment by frouioui** [comment]
@systay, you added the `release notes (needs details)` label. I have modified the release notes for all current versions of `v14` and `v15` to include the known issue, and modified `v16`'s release summary to include a note concerning this fix. I'll let you re-review and unlabel the PR if everything looks good.

File: doc/releasenotes/16_0_0_summary.md [hunk 1]
```diff
@@ -34,6 +34,13 @@ Other aspects of the VReplication copy-phase logic are preserved:
 
  Other phases, catchup, fast-forward, and replicating/"running", are unchanged.
 
+### Important bug fixes
+
+#### Corrupted results for non-full-group-by queries with JOINs
+
+An issue in versions `<= v14.0.3` and `<= v15.0.0` that generated corrupted results for non-full-group-by queries with a JOIN
```
Comment (deepthi) [inline doc/releasenotes/16_0_0_summary.md:41]: Why do we need to specify both <= 14.0.3 and <= 15.0.0?

Comment (frouioui) [reply doc/releasenotes/16_0_0_summary.md:41]: To be explicit with where this issue resides. Might help to know that someone running `v15.0.1` looking to upgrade to `v16.0.0` is not affected by that issue and patch.

Comment (deepthi) [reply doc/releasenotes/16_0_0_summary.md:41]: Is it also an issue on 13.0? Or just 14 & 15?

Comment (frouioui) [reply doc/releasenotes/16_0_0_summary.md:41]: It is an issue only in 14 and 15. We were not supporting this kind of query at all before 14.

**Review by deepthi — APPROVED** [review]
👍 on the release notes changes.

**Pull request merged**